pkg/genericcontrolplane/server.go

```diff
@@ -107,6 +112,27 @@ func CreateServerChain(completedOptions completedServerRunOptions, stopCh <-chan
 		return nil, err
 	}
 
+	// HACK: Adding back namespace controller
```

Comment:
Not sure we need to add the `HACK:` prefix in the comment here since we're making a change on a new file, which is not part of existing Kube code.
As for the commit title, probably it would need the `NEW:` prefix.
@smarterclayton do you confirm ?

Reply:
Now that I'm thinking about it, we could also add this PostStartHook at the KCP layer, which would avoid changing the Kubernetes fork, and would keep the separation between the minimal apiserver itself and the related controllers required for the KCP scenario.
Typically the PostStartHook would be added somewhere [near this line](https://github.com/kcp-dev/kcp/blob/main/pkg/server/server.go#L178)

Reply:
+1 I can do that! that makes way more sense!